ndcube/ndcollection.py

```diff
@@ -0,0 +1,261 @@
+
+    @property
+    def aligned_world_axis_physical_types(self):
+        """
+        Returns the physical types of the aligned axes of an ND object in the collection.
+
+        If there are no aligned axes, returns None.
+
+        """
+        if self.aligned_axes is None:
+            return None
+        else:
+            axis_types = np.array(self[self._first_key].world_axis_physical_types)
+            return tuple(axis_types[np.array(self.aligned_axes[self._first_key])])
+
+    def __getitem__(self, item):
+        # There are two ways to slice:
+        # by key or sequence of keys, i.e. slice out given cubes in the collection, or
+        # by typical python numeric slicing API,
+        # i.e. slice the each component cube along the aligned axes.
+
+        # If item is single string, slicing is simple.
+        if isinstance(item, str):
+            return super().__getitem__(item)
+
+        # If item is not a single string...
+        else:
+            # If item is a sequence, ensure strings and numeric items are not mixed.
+            item_is_strings = False
+            if isinstance(item, collections.abc.Sequence):
+                item_strings = [isinstance(item_, str) for item_ in item]
+                item_is_strings = all(item_strings)
+                # Ensure strings are not mixed with slices.
+                if (not item_is_strings) and (not all(np.invert(item_strings))):
+                    raise TypeError("Cannot mix keys and non-keys when indexing instance.")
+
+            # If sequence is all strings, extract the cubes corresponding to the string keys.
+            if item_is_strings:
+                new_data = [self[_item] for _item in item]
+                new_keys = item
+                new_aligned_axes = tuple([self.aligned_axes[item_] for item_ in item])
+
+            # Else, the item is assumed to be a typical slicing item.
+            # Slice each cube in collection using information in this item.
+            # However, this can only be done if there are aligned axes.
+            else:
+                if self.aligned_axes is None:
+                    raise IndexError("Cannot slice unless collection has aligned axes.")
+                else:
+                    # Derive item to be applied to each cube in collection and
+                    # whether any aligned axes are dropped by the slicing.
+                    collection_items, new_aligned_axes = self._generate_collection_getitems(item)
+                    # Apply those slice items to each cube in collection.
+                    new_data = [self[key][tuple(cube_item)]
+                                for key, cube_item in zip(self, collection_items)]
+                    # Since item is not strings, no cube in collection is dropped.
+                    # Therefore the collection keys remain unchanged.
+                    new_keys = list(self.keys())
```

Comment:
The else is redundant.
```suggestion
                # Derive item to be applied to each cube in collection and
                # whether any aligned axes are dropped by the slicing.
                collection_items, new_aligned_axes = self._generate_collection_getitems(item)
                # Apply those slice items to each cube in collection.
                new_data = [self[key][tuple(cube_item)]
                            for key, cube_item in zip(self, collection_items)]
                # Since item is not strings, no cube in collection is dropped.
                # Therefore the collection keys remain unchanged.
                new_keys = list(self.keys())
```